app/feature/feature-choose-tier/src/main/kotlin/com/hedvig/android/feature/change/tier/data/GetCurrentContractDataUseCase.kt

```diff
@@ -44,19 +45,49 @@ internal class GetCurrentContractDataUseCaseImpl(
         logcat(ERROR) { "Tried to start Change Tier flow but got null active contract" }
         raise(ErrorMessage("Tried to start Change Tier flow but got null active contract"))
       } else {
-        val deductible = dataResult.currentAgreement.deductible?.let {
-          Deductible(
-            deductibleAmount = UiMoney.fromMoneyFragment(it.amount),
-            deductiblePercentage = it.percentage,
-            description = it.displayText,
+        if (dataResult.upcomingChangedAgreement!=null) {
+          val deductible = dataResult.upcomingChangedAgreement.deductible?.let {
+            Deductible(
+              deductibleAmount = UiMoney.fromMoneyFragment(it.amount),
+              deductiblePercentage = it.percentage,
+              description = it.displayText,
+            )
+          }
+          CurrentContractData(
+            currentExposureName = dataResult.exposureDisplayName,
+            currentDisplayPremium = UiMoney.fromMoneyFragment(dataResult.upcomingChangedAgreement.premium),
+            deductible = deductible,
+            productVariant = dataResult.upcomingChangedAgreement.productVariant.toProductVariant(),
+            displayItems = dataResult.upcomingChangedAgreement.displayItems.map {
+              ChangeTierDeductibleDisplayItem(
+                displayTitle = it.displayTitle,
+                displaySubtitle = it.displaySubtitle,
+                displayValue = it.displayValue,
+              )
+            }
+          )
+        } else {
+          val deductible = dataResult.currentAgreement.deductible?.let {
+            Deductible(
+              deductibleAmount = UiMoney.fromMoneyFragment(it.amount),
+              deductiblePercentage = it.percentage,
+              description = it.displayText,
+            )
+          }
+          CurrentContractData(
+            currentExposureName = dataResult.exposureDisplayName,
+            currentDisplayPremium = UiMoney.fromMoneyFragment(dataResult.currentAgreement.premium),
+            deductible = deductible,
+            productVariant = dataResult.currentAgreement.productVariant.toProductVariant(),
+            displayItems = dataResult.currentAgreement.displayItems.map {
+              ChangeTierDeductibleDisplayItem(
+                displayTitle = it.displayTitle,
+                displaySubtitle = it.displaySubtitle,
+                displayValue = it.displayValue,
+              )
+            }
```

Comment:
Is it possible at all to perhaps here merge these two branches together by doing
```
val agreement = dataResult.upcomingChangedAgreement ?: dataResult.currentAgreement
```
And letting everything below it stay the same?
Do they share the same type for this to be possible? If not ignore this.